.github/workflows/lint.yml

```diff
@@ -93,7 +93,10 @@ jobs:
           github_token: ${{ secrets.github_token }}
           reporter: github-pr-check
           level: warning
-          flags: --linelength=120 --exclude=java/src/main/native/*.c
+          flags: --linelength=120
+            --exclude=java/src/main/native/*.c
+            --exclude=onnxruntime/core/mlas/inc/*
+            --exclude=onnxruntime/core/mlas/lib/*
```

Comment:
Disables the Github Lint C++ workflow on MLAS files.

MLAS uses its own formatting style, and the excessive linter warnings caused reviewdog to throw an error that prevented lint checks from passing:
![lint_cpp_mlas_too_many_results](https://github.com/microsoft/onnxruntime/assets/19691973/87ea5cdb-3aa6-426e-bbf3-0ef98c1ddb09)

The ideal fix would be to incorporate MLAS's specific linting rules into our workflows, but that can be done in a separate PR.